localflavor/bg/validators.py

```diff
@@ -0,0 +1,68 @@
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+from .utils import get_egn_birth_date
+
+
+def egn_validator(egn):
+    '''
+    Check Bulgarian unique citizenship number (EGN) for validity
+    More details https://en.wikipedia.org/wiki/Unique_citizenship_number
+    Full information in Bulgarian about algoritm is available here
+    http://www.grao.bg/esgraon.html#section2
+
+    '''
+    def check_checksum(egn):
+        _weights = (2, 4, 8, 5, 10, 9, 7, 3, 6)
+        try:
+            checksum = sum(weight * int(digit) for weight, digit in zip(_weights, egn))
+            return int(egn[-1]) == checksum % 11 % 10
```

Comment:
Is it possible to express this a little more verbosely? I'm just thinking it would be easier to read if it didn't have the double modulus operator. This is just a suggestion though as there could be specific reason you're doing it like this.